### DISCUSSION TITLE
Failed to link, Invalid Argument

### DISCUSSION THREAD

**Discussion by jokem59** [libbpf/libbpf-bootstrap#207] [Q&A] [answered]
I've pulled down the repo on my local machine and am attempting to make the the examples from `local/home/joe/dev/libbpf-bootstrap/examples/c`.

The output of `make`:
```
  BPF      .output/minimal_legacy.bpf.o
libbpf: relocation against STT_SECTION in non-exec section is not supported!
Error: failed to link '.output/minimal_legacy.tmp.bpf.o': Invalid argument (22)
make: *** [.output/minimal_legacy.bpf.o] Error 234
make: *** Deleting file `.output/minimal_legacy.bpf.o'
```

I haven't edited the Makefile at all.  What could be causing the Invalid Argument error?

**Comment by jokem59** [comment, marked as answer]
I found this e-mail thread: https://lore.kernel.org/bpf/43d9bd1c-fba6-f653-7374-cc593f01bc72@fb.com/

It was pointed out that the `.eh_frame` section shouldn't exist in these bpf object files.  But examining my object files showed its existence:
```
readelf -S .output/minimal_legacy.tmp.bpf.o
[17] .eh_frame         PROGBITS         0000000000000000  00000af0
```

Currently using Clang 11.X, and passing `-fno-asynchronous-unwind-tables` into the clang command solved this for me!

**Reply by Ranler** [reply]
this patch:
```
diff --git a/examples/c/Makefile b/examples/c/Makefile
index faefebc..a6131b5 100644
--- a/examples/c/Makefile
+++ b/examples/c/Makefile
@@ -107,6 +107,7 @@ $(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(VMLINUX) | $(OUTPUT)
        $(call msg,BPF,$@)
        $(Q)$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(ARCH)               \
                     $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES)                    \
+                    -fno-asynchronous-unwind-tables                          \
                     -c $(filter %.c,$^) -o $(patsubst %.bpf.o,%.tmp.bpf.o,$@)
        $(Q)$(BPFTOOL) gen object $@ $(patsubst %.bpf.o,%.tmp.bpf.o,$@)
```

**Reply by anakryiko** [reply]
@Ranler feel free to send the patch to add this permanently to Makefile. BTW, do we need `-fno-asynchronous-unwind-tables` or `-fno-unwind-tables` would be enough?
